plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/completion/Completions.kt

```diff
@@ -18,12 +18,14 @@ internal object Completions {
                 complete(factory.callableContributor(0), positionContext)
                 complete(factory.classifierContributor(0), positionContext)
                 complete(factory.packageCompletionContributor(1), positionContext)
+                complete(factory.declarationFromUnresolvedNameContributor(1), positionContext)
             }
 
             is FirTypeNameReferencePositionContext -> {
                 complete(factory.classifierContributor(0), positionContext)
                 complete(factory.keywordContributor(1), positionContext)
                 complete(factory.packageCompletionContributor(2), positionContext)
+                complete(factory.declarationFromUnresolvedNameContributor(1), positionContext)
```

Comment:
Why do we need `declarationFromUnresolvedNameContributor` on completion in type reference?

Reply:
This is to complete an unresolved class name.

Reply:
This is for  variable and function completion. For example, with `val i<caret>`, the fake file has `val  iX.f` and hence `iX` is treated as a type reference.